Adds surgery to remove surgically attached chainsaws

PKPenguin321 opened the pull request:
Also fixes the indentation on the surgery to add chainsaws. It was a bunch of spaces, now it's neat tabs.

Allows you to remove chainsaws that have been surgically attached. The only step is using a saw. You can use the standard surgical saw and all normal ghetto analogues (such as armblades or cleavers), as well as _other chainsaws and arm-mounted chainsaws._ Enjoy your chainsaw arm highlander.

Fixes #12785

PKPenguin321 committed:
adds removing chainsaws + indentation cleanup

changes indenting on chainsaw augmentation from a bunch of spaces to nice and neat tabs
adds a surgery to remove surgically attached chainsaws

optimumtact commented on code/modules/surgery/limb augmentation.dm:
```diff
@@ -77,31 +77,62 @@
+		qdel(V)
+		new /obj/item/weapon/twohanded/required/chainsaw(get_turf(target))
+		user.visible_message("[user] carefully saws [target]'s arms free of the chainsaw.", "<span class='notice'>You remove the chainsaw.</span>")
+		return 1
```
this means you will remove only one chainsaw at a time as the loop will only execute once in this case.

You may have wanted to do that in which case it's fine, just thought I'd flag it for you.

PKPenguin321 replied:
yeah its intentional

PKPenguin321 committed:
changes messages to be more clear

I forgot to change the messages to make them more clear, they were previously "checking for a chainsaw" since the way i checked for chainsaws was much more hacky in the original version of this code

lzimann commented on code/modules/surgery/limb augmentation.dm:
```diff
@@ -77,31 +77,62 @@
+		user << "<span class='warning'>You can't fit the chainsaw in while [target]'s hands are full!</span>"
+		return 0
+	else
+		user.visible_message("[user] finshes installing the chainsaw!", "<span class='notice'>You install the chainsaw.</span>")
```
Small typo, "finshes".

PKPenguin321 replied:
good catch, i wish you had noticed this before i made my last commit

PKPenguin321 committed:
fixes critical bugs

finishes -> finshes
arms -> arm

sorry for clogging commits :(

Firecage commented:
THank god PK. Do you know the hell of when you go braindead...and a roboticist attaches saws to BOTH hands.

optimumtact commented:
lol

KorPhaeron commented:
>THank god PK. Do you know the hell of when you go braindead...and a roboticist attaches saws to BOTH hands.

I think this is a pretty good reason to close the PR

Supermichael777 commented:
>THank god PK. Do you know the hell of when you go braindead...and a roboticist attaches saws to BOTH hands.

![Knifehandst](http://static.tvtropes.org/pmwiki/pub/images/rsz_razor_fist_william_scott_3750.jpg)

PKPenguin321 commented:
i just realized, if this and #12739 both get merged you'll be able to cut off your own chainsaw hand with your own chainsaw hand

Shapsy commented:
You realize you'd still have no arm right

PKPenguin321 commented:
>Dorsisdwarf added the Ready for Review/Merge label 8 days ago

>**8 days ago**

MrStonedOne commented:
Please make a changelog,

You can add it to the PR's body rather then as a file now, see here: https://tgstation13.org/phpBB/viewtopic.php?f=5&t=5157

duncathan commented on code/modules/surgery/limb augmentation.dm:
```diff
@@ -77,31 +77,62 @@
 
 /datum/surgery_step/chainsaw/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
-        user.visible_message("[user] begins to install the chainsaw onto [target].", "<span class='notice'>You begin to install the chainsaw onto [target]...</span>")
+	user.visible_message("[user] begins to install the chainsaw onto [target].", "<span class='notice'>You begin to install the chainsaw onto [target]...</span>")
```
Wtf is with the whitespace changes

Jordie0608 replied:
Whitespaces were spaces beforehand, changed to tabs

duncathan replied:
Oh good this is fine then
